docs/protocol/interchain-gas-payment.mdx

```diff
@@ -1,59 +1,73 @@
-
-:::
-
 Successful interchain messages require two transactions; one on the origin chain to send the message, and one on the destination chain to deliver the message.
 
 For convenience, Hyperlane provides an onchain interface on the origin chain that allows message senders to pay the [relayer](./agents/relayer.mdx) to deliver a message on the destination chain. This payment is called an interchain gas payment.
 
 ```mermaid
 flowchart LR
-    subgraph Origin
+    subgraph Origin Chain
       Sender
       M_O[(Mailbox)]
       IGP[InterchainGasPaymaster]
-      Sender -- "id = dispatch(message)" --> M_O
-      Sender -- "payForGas(id)\n{value: payment}" --> IGP
+      Sender -- "dispatch(...)\n{value}" --> M_O
+      M_O -- "postDispatch(..., gasLimit)\n{value}" --> IGP
     end
 
     Relayer((Relayer))
 
-    M_O -. "emit Dispatch(message)" .-> Relayer
-    IGP -.- Relayer
+    M_O -. "indexing" .-> Relayer
+    IGP -. "value" .- Relayer
 
-    subgraph Destination
+    subgraph Destination Chain
       M_D[(Mailbox)]
+      M_D -- "handle(...)" --> Recipient
     end
 
-    Relayer -- "process(message)" --> M_D
-
-    style Sender fill:#efab17
-
-    click IGP https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/igps/InterchainGasPaymaster.sol
+    Relayer -- "process{gasLimit}(...)" --> M_D
 ```
 
 ## InterchainGasPaymasters
 
-Interchain gas payments are facilitated by `InterchainGasPaymaster` (IGP) smart contract contracts.
+Interchain gas payments are facilitated by `InterchainGasPaymaster` (IGP) smart contracts.
 
-These contracts expose the [InterchainGasPaymaster interface](../reference/hooks/interchain-gas.mdx), which allows users to pay the [relayer](./agents/relayer.mdx) with native tokens on the origin chain to cover the costs of delivering a message on the destination chain.
+These contracts expose the [InterchainGasPaymaster interface](../reference/hooks/interchain-gas.mdx), which allow users to pay the [relayer](./agents/relayer.mdx) with native tokens on the origin chain to cover the costs of delivering a message on the destination chain.
 
 Each `InterchainGasPaymaster` contract corresponds to exactly one relayer. You can find addresses of the Abacus Works' relayer under [addresses](../reference/contract-addresses.mdx)
 
+### Gas limit
+
+The gas limit is the cost of calling `handle` on the destination chain for a given message. This can vary depending on message content and the logic of the handler.
+
+### Gas overhead
+
+The gas overhead is set as part of the destination gas configuration, along with the `gasOracle`:
+
+```solidity
+struct DomainGasConfig {
+    IGasOracle gasOracle;
+    uint96 gasOverhead;
+}
+```
+
+This corresponds to the operational cost of processing a message on the destination chain.
+
+:::note
+You should ensure the `gasOverhead` sufficiently covers the range of ISMs on your destination chain.
```

Comment:
maybe covers the "set" instead of "range" here

Reply:
ah yes set works better here, will try and hit up on my next docs PR